control-plane/api-gateway/common/resources.go

```diff
@@ -116,6 +116,7 @@ type ResourceMap struct {
 	httpRouteGateways     map[api.ResourceReference]*httpRoute
 	gatewayResources      map[api.ResourceReference]*resourceSet
 	externalFilters       map[corev1.ObjectReference]client.Object
+	gatewayPolicies       map[corev1.ObjectReference]*v1alpha1.GatewayPolicy
```

Comment:
I think we might need to use `api.ResourceReference` as the key here so that we can reference the policy by the listener it applies to, what do you think?

Reply:
I'll give it a try. I think I initially thought it was 1to1 and wasn't accounting for the listeners.